decidim-verifications/app/controllers/decidim/verifications/sms/authorizations_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Verifications
+    module Sms
+      class AuthorizationsController < ApplicationController
+        helper_method :authorization
+
+        before_action :load_authorization
+
+        def new
+          enforce_permission_to :create, :authorization, authorization: @authorization
+
+          @form = MobilePhoneForm.new
+        end
+
+        def create
+          enforce_permission_to :create, :authorization, authorization: @authorization
+
+          @form = MobilePhoneForm.from_params(params.merge(user: current_user))
+
+          PerformAuthorizationStep.call(@authorization, @form) do
+            on(:ok) do
+              flash[:notice] = t("authorizations.create.success", scope: "decidim.verifications.sms")
+              authorization_method = Decidim::Verifications::Adapter.from_element(authorization.name)
+              redirect_to authorization_method.resume_authorization_path
+            end
+            on(:invalid) do
+              flash.now[:alert] = t("authorizations.create.error", scope: "decidim.verifications.sms")
+              render :new
+            end
+          end
+        end
+
+        def edit
+          enforce_permission_to :update, :authorization, authorization: @authorization
+
+          @form = ConfirmationForm.from_params(params)
+        end
+
+        def update
+          enforce_permission_to :update, :authorization, authorization: @authorization
+
+          @form = ConfirmationForm.from_params(params)
+
+          ConfirmUserAuthorization.call(@authorization, @form) do
+            on(:ok) do
+              flash[:notice] = t("authorizations.update.success", scope: "decidim.verifications.sms")
+              redirect_to decidim_verifications.authorizations_path
+            end
+
+            on(:invalid) do
+              flash.now[:alert] = t("authorizations.update.error", scope: "decidim.verifications.sms")
+              render :edit
+            end
+          end
+        end
+
+        private
+
+        # rubocop:disable Naming/MemoizedInstanceVariableName
+        def authorization
+          @authorization_presenter ||= AuthorizationPresenter.new(@authorization)
```

Comment:
Hi @oriolgual, this method seems to be copy&pasted from other verification methods (`id_documents` or `postal_letter`) but in the SMS verification method you haven't defined an `AuthorizationPresenter` class as they did.

I guess that this is not failing because `@authorization` is used instead of this method, and it is not used as a helper method in the view, but I'm sure that it will fail when someone tries to use it.

Reply:
Hi @oriolgual @leio10 has this comment been taken into account?
I see the PR has already been merged, but don't know if this was fixed.

Reply:
Hi @tramuntanal, nope, I've found the problem after the PR's merge. Probably, a PR with a fix for this would be more useful, but I can't invest time on that now :(

Reply:
Fixed https://github.com/decidim/decidim/pull/4858